src/lib/isUrl.ts

```diff
@@ -1,4 +1,8 @@
 export default function isUrl(text: string) {
+  if (text.match(/\n/)) {
```

Comment:
Seems like `new URL` strips newlines and considers whatever is left, so we need a special case here.